src/scipp/interpolate/__init__.py

```diff
@@ -150,7 +156,12 @@ def func(xnew: Variable, *, midpoints=False) -> DataArray:
                            fill_value=fill_value,
                            **kwargs)
         x_ = _as_interpolation_type(_midpoints(xnew, dim) if midpoints else xnew)
-        ynew = array(dims=da.dims, unit=da.unit, values=f(x_.values))
+        sizes = da.sizes
+        sizes[dim] = x_.sizes[dim]
+        # ynew is created in this manner to allow for creation of structured dtypes,
+        # which is not possible using scipp.array
+        ynew = empty(sizes=sizes, unit=da.unit, dtype=da.dtype)
```

Comment:
Could we use `interp1d`'s `copy` arg to avoid allocating the output like this? I.e.
```.py
ynew = da.values.copy()
inter.interp1d(x=..., y=ynew, copy=False, ...)
```
Not sure if this would modify `x`, though.

Reply:
The length along `dim` changes with the interpolation so copying is not an option.